components/log-viewer-webui/client/webpack.config.js

```diff
@@ -0,0 +1,82 @@
+import HtmlWebpackPlugin from "html-webpack-plugin";
+import MiniCssExtractPlugin from "mini-css-extract-plugin";
+import * as path from "node:path";
+import {fileURLToPath} from "node:url";
+
+
+const filename = fileURLToPath(import.meta.url);
+const dirname = path.dirname(filename);
+
+const isProduction = "production" === process.env.NODE_ENV;
+
+const stylesHandler = isProduction ?
+    MiniCssExtractPlugin.loader :
+    "style-loader";
+
+const plugins = [
+    new HtmlWebpackPlugin({
+        template: path.resolve(dirname, "public", "index.html"),
+    }),
+];
+
+const config = {
+    devtool: isProduction ?
```

Comment:
Is the `devServer` config to be added in a future PR?

Reply:
Maybe I read the docs incorrectly, but it seemed like the defaults worked?

Reply:
Right, the defaults do work, but do we want to be explicit about which port to use? Also, similar to HMR, do we want to enable "Fast Refresh" with https://github.com/pmmmwh/react-refresh-webpack-plugin ?

Reply:
* I think the port defaults to `'auto'` which seems like it would be more flexible than specifying it explicitly (and the CLI prints out what it chose).
* We could add `open` if you like (I prefer not automatically opening the browser, but that's just me).
* What does that plugin do compared to the default dev server?

Reply:
> I think the port defaults to 'auto' which seems like it would be more flexible than specifying it explicitly (and the CLI prints out what it chose).

Gotcha. Let's leave it as is then.

> We could add open if you like (I prefer not automatically opening the browser, but that's just me).

I don't have a preference so I'm fine leaving it as `false`.

> What does that plugin do compared to the default dev server?

It's for preserving states of any modified React components. e.g. Say we are modifying a button in a modal window, with conventional HMR the modal window will be closed when the page reloads. With Fast Refresh, the modal window will remain open without us clicking again in the browser. Another example is with Fast Refresh, any existing text input is preserved in a text field.

Reply:
Gotcha, thanks.